src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,202 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        this.elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+    }
 
+    public boolean add(T e) {
+        add(e, elementData, size);
```

Comment:
elementData and size are class variables. There is no reason to pass them as arguments to the private method. You can use them in any method inside this class.